src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp/src/components/api-authorization/AuthorizeService.js

```diff
@@ -0,0 +1,323 @@
+﻿import { UserManager, WebStorageStateStore } from 'oidc-client';
+import { ApplicationPaths, ApplicationName } from './ApiAuthorizationConstants';
```

Comment:
I don't know `oidc-client` well enough to really assess the correctness and completeness of all these flows, but it looks totally plausible! It's great that you've been able to handle such a range of scenarios here.